libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp

```diff
@@ -28,8 +33,13 @@ void RCInput_ZYNQ::init()
     if (mem_fd == -1) {
         AP_HAL::panic("Unable to open /dev/mem");
     }
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ALTERA
+    pulse_input = (uint32_t*) mmap(0, 0x200000, PROT_READ|PROT_WRITE,
+                                                      MAP_SHARED, mem_fd, RCIN_ZYNQ_PULSE_INPUT_BASE);
+#else
     pulse_input = (volatile uint32_t*) mmap(0, 0x1000, PROT_READ|PROT_WRITE,
                                                       MAP_SHARED, mem_fd, RCIN_ZYNQ_PULSE_INPUT_BASE);
+#endif
```

Comment:
add a define above for the size: `RCIN_ZYNQ_PULSE_INPUT_BASE`, so you don't need the ifdef here.